client/chat/action-creators.ts

```diff
@@ -15,7 +26,28 @@ export function joinChannel(channel: string): ThunkAction {
     })
     dispatch({
       type: '@chat/joinChannel',
-      payload: fetchJson<void>(apiUrl`chat/${channel}`, { method: 'POST' }),
+      payload: fetchJson<void>(apiUrl`chat/${channel}`, { method: 'POST' }).catch(err => {
```

Comment:
I went with `abortableThunk` approach, but I still wanted to handle the errors in the action-creator, instead of the component itself. The reason for this is that I assume we'll eventually be dispatching this action from multiple places and we'd need to handle these errors in all of them otherwise.

Reply:
I would very much recommend against that, error handling is almost *always* specific to the UI in which the error was generated. If you're trying to join a channel through a dialog, it might make sense to show errors in the dialog. If you're trying to join by clicking a link in another channel, we probably want to show a snackbar or bring up a dialog, etc.

I handled some errors in action creators for matchmaking because it was very hard to avoid at the time, but in practice it's sort of terrible and has led to other problems with hot reloading, etc. because of action creators depending on TSX files.